cloud/gcp/deploy/up.go

```diff
@@ -266,6 +250,41 @@ func (d *DeployServer) Up(request *deploy.DeployUpRequest, stream deploy.DeployS
 			}
 		}
 
+		// Deploy all buckets
+		buckets := map[string]*storage.CloudStorageBucket{}
+		for _, res := range request.Spec.Resources {
+			switch b := res.Config.(type) {
+			case *deploy.Resource_Bucket:
+				buckets[res.Name], err = storage.NewCloudStorageBucket(ctx, res.Name, &storage.CloudStorageBucketArgs{
+					StackID:  stackID,
+					Bucket:   b.Bucket,
+					Location: details.Region,
+				}, defaultResourceOptions)
+				if err != nil {
+					return err
+				}
+
+				for idx, notification := range b.Bucket.Notifications {
+					// Get the deployed execution unit
+					unit, ok := execs[notification.GetExecutionUnit()]
+					if !ok {
+						return fmt.Errorf("invalid execution unit %s given for topic subscription", notification.GetExecutionUnit())
+					}
+
+					_, err = storage.NewCloudStorageNotification(ctx, fmt.Sprintf("notification-%d-%s", idx, res.Name), &storage.CloudStorageNotificationArgs{
+						StackID:  stackID,
+						Location: details.Region,
+						Bucket:   buckets[res.Name],
+						Config:   notification.Config,
+						Function: unit,
+					}, defaultResourceOptions)
+					if err != nil {
+						return err
+					}
+				}
+			}
```

Comment:
Is this safe? Can we guarantee that the order of notifications is deterministic?

Probably best to include more specific unique information, like the target name instead of the index.